utkdigitalinitiatives/utk-archivesspace-local
# As 30

**JIRA Ticket**: [AS 30](https://jira.lib.utk.edu/browse/AS-30)


# What does this Pull Request do?

Accession date does not show on the search listing returned when performing a browse.

The ticket states In the Accession view, display date, accession identifier/number, and processing status (cataloged, uncataloged, processed, and unprocesses).

# What's new?

Add in a fl parameter to solr request. Place if, then statements to produce a sortable accession date in the accession view.


# How should this be tested?
vagrant destroy

vagrant up

vagrant ssh and switch to this feature branch in UTKdrupal, AS-30
$ cd
$ git clone https://github.com/utkdigitalinitiatives/utk-archivesspace-local.git
$ cd utk-archivesspace-local
$ git fetch
$ git checkout AS-30
$ sudo cp -rf * /vagrant/archivesspace/plugins/local/

For me, archive space did not download and install properly with the vagrant up process. So


$ cd /tmp

$ sudo curl -L -O https://github.com/archivesspace/archivesspace/releases/download/v2.5.0/archivesspace-v2.5.0.zip
$ unzip -d /vagrant/ archivesspace-v2.5.0.zip
$ cd /vagrant/archivesspace/
$ ./archivesspace.sh start

Wait a couple minutes. Literally, take out a stop watch and note that the third hand 'clicks' 120 times. Then you should be ready to go on. Unless your computer is ancient, then wait 10 minutes.

Open up your favorite browser on your base machine and type http://localhost:8080/ , then press enter.

Login with admin/admin as that is the username/password.

Click on the 'System' dropdown menu at the top right corner and choose, 'Manage Repositories'.

Click on the 'Create Repository' button at the top right. 

Enter 'Repository Short Name' as Test, Enter 'Repository Name' as Test, then click the 'Save' button at the top of the 'New Repository' Form.

Along the top navigation bar, click on the drop down 'Create'. Choose 'Background Job' and 'Import Data' to open up the data import form.

You will need to download this file:

https://raw.githubusercontent.com/archivesspace/archivesspace/master/backend/app/exporters/examples/accession/aspace_accession_import_template.csv

Save it in your favorite download directory.

On the 'Import Type' dropdown in the middle of the form, select 'Assessment CSV'.

Click the '+ Add file' button. Navigate to your favorite download directory on your base box's filesystem and select 'aspace_accession_import_template.csv' then hit the 'Open' button.

Push the 'Queue Job' button on the bottom right of the 'New Background Job — Import Data' . And it fails. Since this is the only test data we have for this kind of crap, we have no plans to correct this situation, and it doesn't seem to be all that important anyhow, Then we proceed with our fingers crossed.

OK. Next, what is supposed to happen, is setting up some records other than assessions.

Click the '+ Add file' button. Navigate to the local directory in which you have the project utk-archivesspace-vagrant install. Select all the files in the utk-archivesspace-vagrant/aspace-example-eads folder and the hit the 'Queue Job' button.

Wait for a good long time until the Job Status is Completed. But don't worry, because this job will fail as well. And that's really about as much as we can do here. Just two more tasks.

Click on the Browse dropdown on the top navigation bar. Choose 'Accession's. The form should return without error with the text 'No records found'.

That's all folks. Using our current vagrant setup.

# Interested parties
Tag (@ mention) interested parties or, if unsure, @utkdigitalinitiatives/digital-initiatives-developers
